[JSC] `Function.prototype.toString` should print `set` / `get` for accessor properties

sosukesuzuki opened the pull request:
#### 1ff045b365bb63494783062033d5ef8a0c629cb7
<pre>
[JSC] `Function.prototype.toString` should print `set` / `get` for accessor properties
<a href="https://bugs.webkit.org/show_bug.cgi?id=271946">https://bugs.webkit.org/show_bug.cgi?id=271946</a>

Reviewed by Alexey Shvayka.

`Function.prototype.toString` should output strings like `function get foo() { [native code] }` or `function set foo() { [native code] }` for accessor properties[1].
However, for some properties such as `RegExp.input`, JSC currently outputs `function foo() { [native code] }` as if it were a regular function.
This patch changes `Function.prototype.toString` to output `get` or `set` for those accessor properties.

[1]: <a href="https://tc39.es/ecma262/#sec-function.prototype.tostring">https://tc39.es/ecma262/#sec-function.prototype.tostring</a>

* JSTests/stress/function-toString-for-accessor-properties.js: Added.
(assertNativeGetter):
(assertNativeSetter):
(wellKnownIntrinsicObjects.forEach):
* LayoutTests/imported/w3c/web-platform-tests/html/dom/idlharness.https_include=(Document_Window)-expected.txt:
* LayoutTests/js/basic-strict-mode-expected.txt:
* LayoutTests/js/dom/native-bindings-descriptors-expected.txt:
* LayoutTests/js/dom/native-bindings-descriptors.html:
* LayoutTests/js/script-tests/basic-strict-mode.js:
* LayoutTests/js/script-tests/function-toString-vs-name.js:
* LayoutTests/platform/glib/imported/w3c/web-platform-tests/html/dom/idlharness.https_include=(Document_Window)-expected.txt:
* LayoutTests/platform/gtk/imported/w3c/web-platform-tests/html/dom/idlharness.https-expected.txt:
* LayoutTests/platform/ipad/imported/w3c/web-platform-tests/html/dom/idlharness.https-expected.txt:
* LayoutTests/platform/wpe/imported/w3c/web-platform-tests/html/dom/idlharness.https-expected.txt:
* Source/JavaScriptCore/runtime/JSCustomGetterFunction.cpp:
(JSC::JSCustomGetterFunction::create):
* Source/JavaScriptCore/runtime/JSCustomSetterFunction.cpp:
(JSC::JSCustomSetterFunction::create):

Canonical link: <a href="https://commits.webkit.org/276904@main">https://commits.webkit.org/276904@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/d686be23c1f31eefa7deb7ef89ff38b6ccccb817

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/46026 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/25155 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/48611 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/48697 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/42066 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/29468 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/22570 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/37632 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/46604 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/22241 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/39706 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/18823 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/19638 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/40802 "Passed tests") | [✅ 🛠 wpe-skia](https://ews-build.webkit.org/#/builders/52/builds/4070 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/39259 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/42334 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/41153 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/50483 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/45500 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/21022 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/17516 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/44792 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/22322 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/39864 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/43683 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/10211 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/22681 "Built successfully") | | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/52651 "Built successfully") | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/22016 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/10770 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

shvaikalesh reviewed (approved):
Lovely, thank you Sosuke!

One thing: could you please `Tools/Scripts/test262-runner --release -o test/built-ins/Function/prototype/toString`? We might have a new passing test there to remove from `JSTests/test262/expectations.yaml`.

Commit:
[JSC] `Function.prototype.toString` should print `set` / `get` for accessor properties
https://bugs.webkit.org/show_bug.cgi?id=271946

Reviewed by Alexey Shvayka.

`Function.prototype.toString` should output strings like `function get foo() { [native code] }` or `function set foo() { [native code] }` for accessor properties[1].
However, for some properties such as `RegExp.input`, JSC currently outputs `function foo() { [native code] }` as if it were a regular function.
This patch changes `Function.prototype.toString` to output `get` or `set` for those accessor properties.

[1]: https://tc39.es/ecma262/#sec-function.prototype.tostring

* JSTests/stress/function-toString-for-accessor-properties.js: Added.
(assertNativeGetter):
(assertNativeSetter):
(wellKnownIntrinsicObjects.forEach):
* LayoutTests/imported/w3c/web-platform-tests/html/dom/idlharness.https_include=(Document_Window)-expected.txt:
* LayoutTests/js/basic-strict-mode-expected.txt:
* LayoutTests/js/dom/native-bindings-descriptors-expected.txt:
* LayoutTests/js/dom/native-bindings-descriptors.html:
* LayoutTests/js/script-tests/basic-strict-mode.js:
* LayoutTests/js/script-tests/function-toString-vs-name.js:
* LayoutTests/platform/glib/imported/w3c/web-platform-tests/html/dom/idlharness.https_include=(Document_Window)-expected.txt:
* LayoutTests/platform/gtk/imported/w3c/web-platform-tests/html/dom/idlharness.https-expected.txt:
* LayoutTests/platform/ipad/imported/w3c/web-platform-tests/html/dom/idlharness.https-expected.txt:
* LayoutTests/platform/wpe/imported/w3c/web-platform-tests/html/dom/idlharness.https-expected.txt:
* Source/JavaScriptCore/runtime/JSCustomGetterFunction.cpp:
(JSC::JSCustomGetterFunction::create):
* Source/JavaScriptCore/runtime/JSCustomSetterFunction.cpp:
(JSC::JSCustomSetterFunction::create):

Canonical link: https://commits.webkit.org/276904@main
